lua/gluatest/expectations/negative.lua

```diff
@@ -24,6 +24,18 @@ return function( subject, ... )
     end
     expectations.equal = expectations.eq
 
+	function expectations.haveValuesMatch( comparison )
+		local match_count = 0
+		for k, v in pairs(subject) do
+			if table.HasValue(comparison, v) then
```

Comment:
table.HasValue() has a O(n) run time, so your check runs similar to O(n^2). You can convert this check behavior to O(n) by converting one of the tables to have the values you are going to check for as keys (outside the for loop). Use the index operator for the check.  `if valuesToCheck[myValueToCheck] then ... end`

https://wiki.facepunch.com/gmod/table.HasValue

Reply:
That wouldn't work as you can't have duplicate keys but you can have duplicate values.

Reply:
I see, after those fixes it makes sense to me now.